### PR TITLE
Adds EverDrive 64 printing patch

### PR DESCRIPTION
**Pull request by CrashOveride95**
If anyone wants to review please do, I'm open to edits about what I added to readme

### CODE REVIEW

File: enhancements/README.md [hunk 1]
```diff
@@ -26,6 +26,15 @@ This allows you to draw 3D boxes for debugging purposes.
...
 Use `debug_box_rot` to draw a box rotated in the xz-plane. If you want to draw a box by specifying min and max points, use `debug_box_pos` instead.
 
+## EverDrive 64 3.0 Printing = `ed64printf.patch`
+
+This allows you to print messages to a console on your PC, similar to how original N64 dev hardware functioned. This only works with
+EverDrive 64 3.0 hardware. This patch takes care of initalizing the Everdrive hardware, linking the EverDrive code, and 
+flushing the USB cache for you. There are two types of Printf calls, `ed64Printf` and `ed64PrintfSync`. `ed64PrintfSync` has guaranteed delivery, but will temporarily stop the thread for as long as it takes to deliver the message. Useful for crash messages.
+The EverDrive 64 FTDI chip has a buffer of 128 bytes, so you can only send 128 characters per frame. The flush call is in the 
+main game loop in /game/game_init.c. For compatibility reasons, ed64 printing can be disabled on build by compiling with ED64DBG=NO.
+The PC relies on an external program by `jsdf` called `ed64log`. This program has only been tested to work on MacOS and Linux.
```
Comment (streetster) [inline enhancements/README.md:36]: Is it worth adding a direct link to ed64log, e.g.

```
The PC relies on an external program by `jsdf` called [ed64log](https://github.com/jsdf/ed64log). 
```

for the lazy people :)

File: enhancements/ed64printf.patch [hunk 2]
```diff
@@ -0,0 +1,1354 @@
...
++#define FAT_ERR_OUT_OF_TABLE 109
++#define FAT_ERR_INIT 110
++#define FAT_LFN_BUFF_OVERFLOW 111
++#define FAT_DISK_NOT_READY 112
++#define FAT_ERR_SIZE 113
++#define FAT_ERR_RESIZE 114
++
++#define ERR_FILE8_TOO_BIG 140
++#define ERR_FILE16_TOO_BIG 141
++#define ERR_WRON_OS_SIZE 142
```
Comment (streetster) [inline enhancements/ed64printf.patch:176]: WRON -> WRONG?

File: enhancements/README.md (hunk 1 above)
Comment (CrashOveride95) [reply enhancements/README.md:36]: I didn't know if it would be tolerated to add a link, if that's allowed i'd be fine with it

**Comment by CrashOveride95** [comment]
PR closed to be resubmitted on private repo

**Pull request closed without merging**